components/filter/filter.js

```diff
@@ -496,7 +497,8 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 			switch (type) {
 				case 'd2l-filter-dimension-set': {
 					info.searchType = dimension.searchType;
-					if (dimension.selectAll) info.selectAllIdPrefix = SET_DIMENSION_ID_PREFIX;
+					info.selectionSingle = dimension.selectionSingle;
+					if (dimension.selectAll && !dimension.selectionSingle) info.selectAllIdPrefix = SET_DIMENSION_ID_PREFIX;
```

Comment:
In the `selection-single` case, the select all checkbox and summary are hidden automatically.  But it seems like they still take up a bit of space, and I also want to hide my container with margins anyways